tortoise/__init__.py

```diff
@@ -539,13 +539,23 @@ def _discover_models(cls, models_path: str, app_label: str) -> List[Type[Model]]
             if isinstance(info, str):
                 info = expand_db_url(info)
             client_class = cls._discover_client_class(info.get("engine"))
-            db_params = info["credentials"].copy()
-            db_params.update({"connection_name": name})
-            connection = client_class(**db_params)  # type: ignore
-            if create_db:
-                await connection.db_create()
-            await connection.create_connection(with_db=True)
-            cls._connections[name] = connection
+
+            # If no credentials assume given pool and acquisition function.
```

Comment:
We should split init into 2 parts:
1. init_models
2. connect_to_db
I did some refactoring to make 1 a bit easier, but it isn't completely done.

You don't need to do it, this seems fine, but I would like to get done cleanly before we do a release.

Reply:
I'm fine working to split the two. Are the changes you are referring to in the Sinaso branch you're working on?

Reply:
No, He basically created a long running branch, and I'm taking some of the useful ideas re maintainablility, and also planning to get the subquery features from there.

The init cleanup/refactor is a different thing. I basically want to beak up the `Tortoise.init(...)` into two parts, the `connections` parameter and the `apps` part.
That should be relatively easy, since I did some of the decoupling, but for some reason the `Tortoise.init_models(...)` has a different api. (It was about initialising models in parts)
Unfortunately the entire init flow just doesn't flow nicely, but what to do isn't so obvious. There is also #14 but I havent put more thought into it.

Reply:
Will still try to separate the inits for connections and models, but for now I've updated to pass mypi and ran black. I would be passing flake8, but there's a bunch of Empty descriptions for arguments in functions I've not touched.
```
flake8 tortoise/ examples/ tests/ conftest.py
tortoise/models.py:752:1: DAR002 Empty description: e (3, 3)
tortoise/models.py:752:1: DAR002 Empty description: e (5, 5)
tortoise/models.py:779:1: DAR002 Empty description: e (13, 13)
tortoise/models.py:840:1: DAR002 Empty description: e (2, 2)
tortoise/models.py:840:1: DAR002 Empty description: e (3, 3)
tortoise/models.py:850:1: DAR002 Empty description: e (2, 2)
tortoise/models.py:850:1: DAR002 Empty description: e (3, 3)
tortoise/models.py:860:1: DAR002 Empty description: e (2, 2)
tortoise/models.py:876:1: DAR002 Empty description: e (7, 7)
tortoise/models.py:893:1: DAR002 Empty description: e (7, 7)
tortoise/models.py:908:1: DAR002 Empty description: e (2, 2)
tortoise/models.py:908:1: DAR002 Empty description: e (3, 3)
tortoise/models.py:908:1: DAR002 Empty description: e (4, 4)
tortoise/query_utils.py:358:1: DAR002 Empty description: e (4, 4)
tortoise/queryset.py:94:1: DAR002 Empty description: e (5, 5)
tortoise/utils.py:11:1: DAR002 Empty description: e (2, 2)
tortoise/utils.py:11:1: DAR002 Empty description: e (3, 3)
tortoise/utils.py:22:1: DAR002 Empty description: e (2, 2)
tortoise/utils.py:22:1: DAR002 Empty description: e (3, 3)
tortoise/backends/base/client.py:35:1: DAR002 Empty description: e (0, 0)
tortoise/backends/base/client.py:35:1: DAR002 Empty description: e (1, 1)
tortoise/backends/base/client.py:35:1: DAR002 Empty description: e (2, 2)
tortoise/backends/base/client.py:35:1: DAR002 Empty description: e (3, 3)
tortoise/backends/base/client.py:35:1: DAR002 Empty description: e (4, 4)
tortoise/backends/base/client.py:157:1: DAR202 Excess "Returns" in Docstring: + return
```

Reply:
Er... I'm happy if you ignore those for now. that is errors re parameters for autodoc docstrings. I also don't see anything that is `e` in there. I can see what the last one talks about (kind of a grammar complaint), but the rest make no sense :-(

Reply:
The reason for the Empty Description is that the argument to the function is listed in the docstring, but there is no description associated. 

So, the first and second warnings:
```
tortoise/models.py:752:1: DAR002 Empty description: e (3, 3)
tortoise/models.py:752:1: DAR002 Empty description: e (5, 5)
```
Are due to
``` Python
"""
... 
:param defaults:
:param kwargs:
...
```
Not having descriptions in the docstring following their listing in the following function definition's docstring.
``` Python
    @classmethod
    async def get_or_create(
        cls: Type[MODEL],
        defaults: Optional[dict] = None,
        using_db: Optional[BaseDBAsyncClient] = None,
        **kwargs: Any,
    ) -> Tuple[MODEL, bool]:
        """
        Fetches the object if exists (filtering on the provided parameters),
        else creates an instance with any unspecified parameters as default values.

        :param defaults:
        :param using_db: Specific DB connection to use instead of default bound
        :param kwargs:
        """
        if not defaults:
```
So
... Python
:param defaults: Some default param that does defaulty stuffs.
:param kwargs: The usual suspects of args that aren't nice enough to get their own arg slot.
...
```
Would resolve those.

I think updating the doc checking package when adding requirements may be the cause for these new warnings.

Reply:
Ah, I remember removing kwargs as it made the docs confusing. Yes, downgrading darglint is a good enough idea for now. It is unfortunately difficult to skip linting, as those "skip instructions" end up in the docs which is unacceptable.

I'll have a look at fixing what makes sense and doing a file-level exclude at a later date, but not right now.